google/cloud/storage/tests/async_client_integration_test.cc

```diff
@@ -127,6 +128,43 @@ TEST_F(AsyncClientIntegrationTest, ComposeObject) {
   EXPECT_EQ(*read.object_metadata, *composed);
 }
 
+TEST_F(AsyncClientIntegrationTest, StreamingRead) {
+  auto object_name = MakeRandomObjectName();
+  // Create a relatively large object so the streaming read makes sense.
+  auto constexpr kSize = 32 * 1024 * 1024L;
+  auto const expected_data = MakeRandomData(kSize);
+
+  auto async = AsyncClient();
+  auto insert = async
+                    .InsertObject(bucket_name(), object_name, expected_data,
+                                  gcs::IfGenerationMatch(0))
+                    .get();
+  ASSERT_STATUS_OK(insert);
+  ScheduleForDelete(*insert);
+
+  auto r = async.ReadObject(bucket_name(), object_name).get();
+  ASSERT_STATUS_OK(r);
+  AsyncReader reader;
+  AsyncToken token;
+  std::tie(reader, token) = *std::move(r);
+
+  std::string actual;
+  while (token.valid()) {
+    ReadPayload payload;
+    auto p = reader.Read(std::move(token)).get();
+    ASSERT_STATUS_OK(p);
+    std::tie(payload, token) = *std::move(p);
+    for (auto v : payload.contents()) actual += std::string(v);
+  }
+  EXPECT_EQ(actual, expected_data);
+
+  auto status = async
+                    .DeleteObject(bucket_name(), object_name,
+                                  gcs::Generation(insert->generation()))
+                    .get();
+  EXPECT_STATUS_OK(status);
```

Comment:
Q. Do we need this if we've `ScheduleForDelete(*insert)`?

Reply:
Nope. removed.